HACKING.md

```diff
@@ -0,0 +1,42 @@
+
+The address of the auth server is configured in the `boot.cfg` file in the installation directory of the client. All other servers can have any address because the previous server sends the address to the client.
+
+Now of course these were only 3 servers but shouldn't there be four servers? - Yes there is also the UGC (User generated content) Server. This server was used for things like icons for modular rockets. As of now the UGC server is not implemented in Uchu.
+
+## The repo structure
+As you probably already noticed the Uchu repository is quite complex. Therefore this paragraph will explain what the different folders contain and what they are for.
+
+There are 3 submodules (marked with a blue text color in GitHub) in the repo.
+* [InfectedRose](https://github.com/Wincent01/InfectedRose) - A library to work with LEGO® Universe files
+* [NexusLogging](https://github.com/TheNexusAvenger/Nexus-Logging) - A logging library for .NET
+* [RakDotNet](https://github.com/UchuServer/RakDotNet) - A library for the RakNet protocol that LU uses
+
+Then there are also the folders that contain the actual Uchu code. These are:
+
+* **Uchu.Api** - It contains code for an api with which the Uchu instance can be controlled. There is a [Cli Tool](https://github.com/UchuServer/cli) that uses this api
+* **Uchu.Auth** - This contains the auth server
+* **Uchu.Char** - This contains the char server
+* **Uchu.Core.Test** - Unit Tests for Uchu.Core
+* **Uchu.Core** - Contains stuff that is used all over the project, like logging, io, database access, etc
+* **Uchu.Instance** - The base for all servers that Uchu runs (like auth, char, world)
+* **Uchu.Master** - This contains the main process that starts and controlls all the servers
+* **Uchu.Navigation** - Code for pathfinding (used by the world server)
+* **Uchu.Physics.Test**- Unit Tests for Uchu.Physics
+* **Uchu.Physics** - Code for all kinds of physics calculations (used by the world server)
+* **Uchu.Python** - Python interpreter for scripts
+* **Uchu.Sso** - Code for lcdr's [Single-Sign-On](https://github.com/lcdr/sso_auth)
+* **Uchu.StandardScripts** - C# vserions of server scripts (most of the original lua server scripts are not available)
+* **Uchu.World.Test** - Unit Test for Uchu.World
+* **Uchu.World** - Code for the world server
+
+If you want to contribute to Uchu most likely Uchu.Wolrd is where you want to make your changes as it contains most of the actual game logic.
+
+## Logging
```

Comment:
It might be worth having more sections like this, for things like Game message handlers, Game message sending functions etc. Just generic functions used all over the code base

Reply:
Added some stuff. What do you think of it?

Reply:
spelling mistake, its Uchu.World :D

Reply:
Already fixed, but thank you :)